echo_client.py

```diff
@@ -2,23 +2,27 @@
 import sys
 
 
-msg = sys.argv[1]
+def echo_client(msg):
+    msg = msg.decode('utf-8')
```

Comment:
There's a trap here.  If the incoming message is already a unicode object, then calling 'decode' on it will implicitly encode it to bytes so that it can apply the decode (I know. Messed up, eh?).  That implicit encoding will happen without you being able to control what codec is used, so you might end up getting `UnicodeEncodeError`s even though you are calling `str.decode`.  If you wan to do this, then you should check first that the message is not unicode before decoding it.